File: templates/ios/Sources/Gyo/BridgeRegistry.swift
import Foundation

// MARK: - Bridge Registry

@MainActor
class BridgeRegistry {
    static let shared = BridgeRegistry()
    
    private var handlers: [String: BridgeHandler] = [:]
    
    private init() {}
    
    func initialize() {
        handlers.removeAll()
    }
    
    func register(_ bridgeName: String, handler: BridgeHandler) {
        handlers[bridgeName] = handler
        print("BridgeRegistry: Registered bridge: \(bridgeName)")
    }
    
    func unregister(_ bridgeName: String) {
        handlers.removeValue(forKey: bridgeName)
        print("BridgeRegistry: Unregistered bridge: \(bridgeName)")
    }
    
    func get(_ bridgeName: String) -> BridgeHandler? {
        return handlers[bridgeName]
    }
}

File: plugins/bridge/android/src/main/kotlin/gyo/plugins/bridge/BridgeRegistry.kt
package gyo.plugins.bridge

import android.util.Log
import java.util.concurrent.ConcurrentHashMap

object BridgeRegistry {
    private val handlers = ConcurrentHashMap<String, BridgeHandler>()

    fun initialize() {
        handlers.clear()
    }

    fun register(bridgeName: String, handler: BridgeHandler) {
        handlers[bridgeName] = handler
        Log.d("BridgeRegistry", "Registered bridge: $bridgeName")
    }

    fun unregister(bridgeName: String) {
        handlers.remove(bridgeName)
        Log.d("BridgeRegistry", "Unregistered bridge: $bridgeName")
    }

    fun get(bridgeName: String): BridgeHandler? {
        return handlers[bridgeName]
    }
}
